compiler/codegen/codegen.cpp

```diff
@@ -2132,21 +2177,58 @@ void codegen(void) {
       lastSlash++;
     }
 
-    // copy from that slash onwards into the executableFilename,
-    // saving space for a `\0` terminator
-    if (strlen(lastSlash) >= sizeof(executableFilename)) {
-      INT_FATAL("input filename exceeds executable filename buffer size");
+    // "Executable" name should be given a "lib" prefix in library compilation,
+    // and just the main module name in normal compilation.
```

Comment:
I was expecting that if somebody ran `chpl --library -o blarg.a blarg.chpl`, then we'd create the file `blarg.a` and not `libblarg.a`. I was expecting `chpl --library blarg.chpl` to make `libblarg.a`, though. How would this code differentiate between these two cases?

Reply:
This code only triggers if the executableName hasn't already been set by the user - see [this line](https://github.com/chapel-lang/chapel/pull/10068/files/3ea24c69fa1a595fcd0d209d5780b4738e6a8abf#diff-bb67493ab4303d061942b7df2da31229R2168).  We set executableFilename from the compiler flag directly